testdata/generate.go

```diff
@@ -0,0 +1,245 @@
+// This binary creates all the necessary certificates and private keys required for the Bootz emulator.
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/golang/glog"
+	"go.mozilla.org/pkcs7"
+)
+
+var (
+	vendor             = flag.String("vendor", "", "The name of the vendor to generate self-signed certificates for.")
+	owner              = flag.String("owner", "", "The name of the organization that owns the emulated device.")
+	controlCardSerials = flag.String("serials", "", "Comma-separated list of control card serials to generate OVs for.")
+)
+
+const (
+	// Default values for the Root CA certificates.
+	caCountry  = "US"
+	caProvince = "CA"
+	caLocality = "Mountain View"
+	// Default one year expiry of certs.
+	caExpiry = time.Hour * 24 * 365
+)
+
+type OwnershipVoucher struct {
+	OV OwnershipVoucherInner `json:"ietf-voucher:voucher"`
+}
+
+// Defines the Ownership Voucher format. See https://www.rfc-editor.org/rfc/rfc8366.html.
+type OwnershipVoucherInner struct {
+	CreatedOn                  string `json:"created-on"`
+	ExpiresOn                  string `json:"expires-on"`
+	SerialNumber               string `json:"serial-number"`
+	Assertion                  string `json:"assertion"`
+	PinnedDomainCert           string `json:"pinned-domain-cert"`
+	DomainCertRevocationChecks bool   `json:"domain-cert-revocation-checks"`
+}
+
+// certFromPem converts a PEM-formatted byte slice to an x509 Certificate
+func certFromPem(contents []byte) (*x509.Certificate, error) {
+	pemBlock, _ := pem.Decode(contents)
+	return x509.ParseCertificate(pemBlock.Bytes)
+}
+
+// privateKeyFromPem converts a PEM-formatted byte slice to an RSA Private Key
+func privateKeyFromPem(contents []byte) (*rsa.PrivateKey, error) {
+	pemBlock, _ := pem.Decode(contents)
```

Comment:
this should return a nil, error on error case and then the next line will panic

you need to handle the error